examples/kubectl.gpt

```diff
@@ -0,0 +1,38 @@
+tools: sys.write, kubectl, helm, browser
+
+1. Create a Namespace named vote but do not fail if it already exists
+2. Deploy in the vote namespace the application which yaml specification is available at https://luc.run/vote.yaml
+3. Use a single command to wait for all the Pods in the vote namespace to be ready
+4. Install Traefik ingress controller in kube-system namespace with helm only if it is not already installed in this namespace
+5. Make sure the Traefik Pod is in running status
+6. Wait for the IP address of the traefik Service to be available and save it in the file ./lb-ip.txt
+7. Create the file ./ingress.yaml and make sure it contains the yaml specification of an Ingress resource which exposes the vote-ui Service on vote.LBIP.nip.io and the result-ui Service on result.LBIP.nip.io, first making sure to replace the LBIP placeholders with the content of the file ./lb-ip.txt
+8. Create the Ingress resource specified in ./ingress.yaml
+9. Open a browser on vote.LBIP.nip.io but make sur to replace the LBIP placeholder with the content of lb-ip.txt in this URL first
+
+---
+name: kubectl
+description: use kubectl command to manage k8s resources
+args: arguments: the arguments for the kubectl command
+
+!#/bin/bash
+
+kubectl ${arguments}
+
+---
+name: helm
+description: use helm command to manage k8s charts
+args: arguments: the arguments for the helm command
+
+!#/bin/bash
+
+helm ${arguments}
+
+---
+name: browser
+tools: sys.exec
+args: url: the url to open
+description: open a browser window
+
+You are only in charge of opening a browser window on the requested url
+You can only use the sys.exec tool to open a browser window
```

Comment:
Can you not use our browser tool for this?

https://tools.gptscript.ai/github.com/gptscript-ai/browser

Reply:
I'm not sure, I need to open a window so show the application. Can the browser tool open a window ?

Reply:
Yep! That's what it is designed to do.